src/servers/grpc_server_v2.cc

```diff
@@ -1315,17 +1361,94 @@ InferGRPCToInput(
+      }
+
+      if (io.contents().int64_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "INT64", io.datatype(),
+            byte_size));
+        base = (const void*)io.contents().int64_contents().data();
+        byte_size = io.contents().int64_contents_size() * 8;
+      }
+
+      if (io.contents().uint_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "UINT32", io.datatype(),
+            byte_size));
+        base = (const void*)io.contents().uint_contents().data();
+        byte_size = io.contents().uint_contents_size() * 4;
+      }
+
+      if (io.contents().uint64_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "UINT64", io.datatype(),
+            byte_size));
+        base = (const void*)io.contents().uint64_contents().data();
+        byte_size = io.contents().uint64_contents_size() * 8;
+      }
+
+      if (io.contents().fp32_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "FP32", io.datatype(), byte_size));
+        base = (const void*)io.contents().fp32_contents().data();
+        byte_size = io.contents().fp32_contents_size() * 4;
+      }
+
+      if (io.contents().fp64_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "FP64", io.datatype(), byte_size));
+        base = (const void*)io.contents().fp64_contents().data();
+        byte_size = io.contents().fp64_contents_size() * 8;
+      }
+
+      if (io.contents().byte_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "STRING", io.datatype(),
+            byte_size));
+        std::shared_ptr<std::string> serialized(new std::string(""));
+        serialized_data_map->emplace(io.name(), serialized);
+
+        // Serialize the output tensor strings. Each string is
+        // serialized as a 4-byte length followed by the string itself
+        // with no null-terminator.
+        for (int e = 0; e < io.contents().byte_contents_size(); ++e) {
+          const std::string& element = io.contents().byte_contents().Get(e);
+          uint32_t len{(uint32_t)element.size()};
+          serialized->append(
+              reinterpret_cast<const char*>(&len), sizeof(uint32_t));
+          if (element.size() > 0) {
+            serialized->append(element.c_str(), len);
+          }
+        }
+        base = serialized->c_str();
+        byte_size = serialized->size();
+      }
     }
 
     uint64_t expected_byte_size = 0;
     RETURN_IF_ERR(TRTSERVER_InferenceRequestProviderInputBatchByteSize(
         request_provider, io.name().c_str(), &expected_byte_size));
 
-    if (byte_size != expected_byte_size) {
+    if ((expected_byte_size != 0) && (byte_size != expected_byte_size)) {
```

Comment:
Why? Is this because for STRING expected_byte_size is returned as 0? If so then you should more explicitly skip this check for string.

Reply:
fixed